### PR TITLE
AUTH-164: Issue #189

### PR DESCRIPTION
**Pull request by luizjaccao**
## Description
ISSUE #189 

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review

### CODE REVIEW

File: components/SnackBar/index.d.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import React = require('react');
+import React from 'react';
 
-export interface SnackBarProps<T> {
+export interface SnackBarProps {
     actionTrigger?: {
         title?: string;
-        callback?: React.MouseEventHandler<T>;
+        callback?: React.MouseEventHandler<HTMLButtonElement>;
```
Comment (rapahaeru) [inline components/SnackBar/index.d.ts]: Doubt: `MouseEventHandler` is the correct type to use in a callback (generic)?

Comment (luizjaccao) [reply components/SnackBar/index.d.ts]: Yes, the `onClick` event belongs to `MouseEventHandler`

File: components/Input/index.d.ts [hunk 2]
```diff
@@ -28,13 +28,13 @@ export interface InputProps<T> {
     name?: string;
     validate?: Validate | CustomValidate | Array<Validate | CustomValidate>;
     mask?: Mask | Mask[];
-    onClean?: React.MouseEventHandler<T>;
-    onChange?: React.ChangeEventHandler<T>;
+    onClean?: React.MouseEventHandler<HTMLInputElement>;
+    onChange?: React.ChangeEventHandler<HTMLInputElement>;
```
[the remark this answers is not in the document]
Comment (rapahaeru) [reply components/Input/index.d.ts]: If i understood, this PR implements Types for all components props and Input does not have a `onBlur` prop.

[the remark this answers is not in the document]
Comment (luizjaccao) [reply components/Input/index.d.ts]: u are right!

**Pull request merged**